File: Eunleelee/2024-02/2월 20일/예상 대진표.py
def solution(n,a,b):
    answer = 0
    person = [i for i in range(1, n+1)]
    person[a-1] = 'A'
    person[b-1] = 'B'
    p = 0; turn = 1
    pp = []
    while 1:
        for i in range(int(len(person)/2)):
            if 'A' in person[p:p+2] and 'B' in person[p:p+2]:
                answer = turn
            elif 'A' in person[p:p+2]:
                pp.append('A')
            elif 'B' in person[p:p+2]:
                pp.append('B')
            else:
                pp.append(i+1)
            p += 2
        if answer != 0:
            break
        turn+=1
        person = pp
        pp = []; p = 0
                

    return answer

File: wooseongshin/2월/20일/예상대진표.java
class Solution {
    public int solution(int n, int a, int b) {
        int cnt = 0;

        while (a != b) {
            a = (a + 1) / 2;
            b = (b + 1) / 2;
            cnt++;
        }
        return cnt;
    }
}
